src/tasks.rs

```diff
@@ -119,6 +131,7 @@ pub struct SucceededTask {
     pub started_at: OffsetDateTime,
     #[serde(with = "time::serde::rfc3339")]
     pub finished_at: OffsetDateTime,
+    pub canceled_by: Option<usize>,
```

Comment:
Same, I think you should use `u32` or `u64`, I think `usize` or `isize` is more for memory no?